src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerHostContext.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Manages properties of analyzers (such as registered actions, supported diagnostics) for analyzer host
```

Comment:
Probably need a better name and doc comment here to indicate the real purpose - we basically need to know the lifetime of the host that is executing analyzers. CompilationWithAnalyzers can be used to execute analyzers on a compilation. However, same instance of an analyzer can be used across different CompilationWithAnalyzers, hence we need CompilationWithAnalyzers to take this object that represents the lifetime of the host. How about naming this `CompilationWithAnalyzersHost`?